<launch>
  <arg name="model" default="$(env TURTLEBOT3_MODEL)"/>
  <arg name="map_file" default="$(find robot_cleaner)/maps/map.yaml"/>
  <arg name="gui" default="true"/> 

  <arg name="x_pos" default="-3.0"/>
  <arg name="y_pos" default="1.0"/>
  <arg name="yaw" default="0.0"/>

  <include file="$(find robot_cleaner)/launch/gazebo_sim.launch">
    <arg name="model" value="$(arg model)"/>
    <arg name="gui" value="$(arg gui)"/>
    <arg name="x_pos" value="$(arg x_pos)"/>
    <arg name="y_pos" value="$(arg y_pos)"/>
    <arg name="z_pos" value="0.0"/>
    </include>

  <include file="$(find robot_cleaner)/launch/navigation.launch">
    <arg name="map_file" value="$(arg map_file)"/>
    <arg name="open_rviz" value="true"/>
    
    <arg name="initial_pose_x" value="$(arg x_pos)"/>
    <arg name="initial_pose_y" value="$(arg y_pos)"/>
    <arg name="initial_pose_a" value="$(arg yaw)"/>
  </include>

  <node pkg="robot_cleaner" type="qr_reader_node.py" name="qr_reader_node" output="screen">
    <param name="image_topic" value="/camera/rgb/image_raw"/>
  </node>

  <node pkg="robot_cleaner" type="task_manager_node.py" name="task_manager_node" output="screen">
    <param name="mission" value="$(find robot_cleaner)/config/mission.yaml"/>
    <param name="report_path" value="$(find robot_cleaner)/reports/clean_report.txt"/>
    <param name="qr_service" value="/qr_reader_node/scan"/>
    <param name="room_timeout" value="120.0"/>
    <param name="goal_timeout" value="60.0"/>
    <param name="max_retries" value="1"/>
  </node>

</launch>
